relayer/send_relayed_transaction.go

```diff
@@ -0,0 +1,90 @@
+package relayer
+
+import (
+	"context"
+	"math/big"
+
+	rcommon "github.com/athanorlabs/go-relayer/common"
+	"github.com/athanorlabs/go-relayer/impls/gsnforwarder"
+	rrelayer "github.com/athanorlabs/go-relayer/relayer"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/athanorlabs/atomic-swap/ethereum/block"
+	"github.com/athanorlabs/atomic-swap/ethereum/extethclient"
+	"github.com/athanorlabs/atomic-swap/net/message"
+)
+
+// ValidateAndSendTransaction sends the relayed transaction to the network if it validates successfully.
+func ValidateAndSendTransaction(
+	ctx context.Context,
+	request *message.RelayClaimRequest,
+	ec extethclient.EthClient,
+	forwarderAddress ethcommon.Address,
+) (*message.RelayClaimResponse, error) {
+
+	err := validateClaimRequest(ctx, request, ec.Raw(), forwarderAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	forwarder, domainSeparator, err := getForwarderAndDomainSeparator(ctx, ec.Raw(), forwarderAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce, err := forwarder.GetNonce(&bind.CallOpts{Context: ctx}, request.Swap.Claimer)
```

Comment:
it would be nice to validate the nonce before we send the tx; ie check that the nonce signed was the same as here, otherwise the tx has already been invalidated. the only drawback if we don't is the chance the tx will revert